### PR TITLE
Reduce dist size

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #1911.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7706/

**Comment by huwshimi** [comment]
QA:
- bootstrap to a cloud
- run `make dist`
- run `juju upgrade-gui` with your dist file
- load the gui and play around with it, there should be no console errors and everything should appear/work as per usual

**Comment by jrwren** [comment]
How much reduction?

File: MANIFEST.in [hunk 1]
```diff
@@ -13,4 +13,9 @@ recursive-include jujugui/templates *
 recursive-exclude jujugui/static/gui/src *
 recursive-exclude jujugui/tests *
 recursive-exclude test *
+recursive-exclude jujugui/static/gui/build/app/components test-*.js
 
+# To get only the *min.js files we first remove all js files and then add the
+# min.js files back in
+recursive-exclude jujugui/static/gui/build *.js
+recursive-include jujugui/static/gui/build *min.js
```
Comment (frankban) [inline MANIFEST.in]: Would this reinclude the minified test files? I am sure it wouldn't just worth checking?

**Comment by frankban** [comment]
FWY, the resulting dist goes from 6.2M to 2.2M.

**Comment by frankban** [comment]
QA in gijoe is ok.

**Comment by hatched** [comment]
It appears that there are a number of 400's when using the new tiny dist with bb. It is loading the non minified versions when using `run`. Before this lands I'd like to have the changes to bb landed so that we don't create a situation where the code is non-releasable. It might be worth holding off until after 2.6.1.  

```
(index):55 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/version.js 
(index):307 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/react-with-addon…&app/assets/javascripts/prism.js&app/assets/javascripts/prism-languages.js 
(index):308 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/yui/yui/yui.js&app/assets/javascripts/yui/loader/loader.js&app/assets/javascripts/d3.js 
(index):309 GET http://10.211.55.8:6543/2.6.0/combo?modules.js 
(index):310 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/js-macaroon.js 
(index):311 GET http://10.211.55.8:6543/2.6.0/combo?app/state/state.js&app/user/user.js&app…ujulib/reconnecting-websocket.js&app/jujulib/urls.js&app/jujulib/bakery.js 
(index):307 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/react-with-addon…&app/assets/javascripts/prism.js&app/assets/javascripts/prism-languages.js 
(index):308 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/yui/yui/yui.js&app/assets/javascripts/yui/loader/loader.js&app/assets/javascripts/d3.js 
(index):309 GET http://10.211.55.8:6543/2.6.0/combo?modules.js 
(index):310 GET http://10.211.55.8:6543/2.6.0/combo?app/assets/javascripts/js-macaroon.js 
(index):311 GET http://10.211.55.8:6543/2.6.0/combo?app/state/state.js&app/user/user.js&app…ujulib/reconnecting-websocket.js&app/jujulib/urls.js&app/jujulib/bakery.js 
```

File: MANIFEST.in (hunk 1 above)
Comment (huwshimi) [inline MANIFEST.in]: You're right! I've moved the line to exclude the test files to the end.

**Comment by huwshimi** [comment]
@frankban do you think we should see if we can have second dist command that creates the dist with the non-minified files as well?

@hatched let's hold off landing this then. Do you think you could create an issue for bb?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7712/

**Comment by huwshimi** [comment]
I have fixed a couple of loading issues on the GUI side and have created CanonicalLtd/pull/1597 to allow the storefront to handle the minified-only dist.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7738/

**Comment by hatched** [comment]
QA appeared to be ok with the most recent modifications however there were other asset related issues that prevented saying for sure if it was ok. This will be blocked until after those are resolved.

**Comment by hatched** [comment]
Thanks @huwshimi qa ok!

The image assets in css are causing issues but those are already being tackled by you :)

**Comment by hatched** [comment]
:shipit: 

Love the new size :)
```
2177811 Jun  6 11:34 jujugui-2.7.3.tar.bz2
6373202 Jun  6 11:33 jujugui-2.7.3.tar.bz2.old
```

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**